lib/src/common/repos_service.dart

```diff
@@ -181,13 +181,15 @@ class RepositoriesService extends Service {
   /// Returns true if it was successfully deleted.
   ///
   /// API docs: https://developer.github.com/v3/repos/#delete-a-repository
-  Future<void> deleteRepository(RepositorySlug slug) async {
+  Future<bool> deleteRepository(RepositorySlug slug) async {
     ArgumentError.checkNotNull(slug);
-    await _github.request(
-      'DELETE',
-      '/repos/${slug.fullName}',
-      statusCode: StatusCodes.NO_CONTENT,
-    );
+    return _github
+        .request(
+          'DELETE',
+          '/repos/${slug.fullName}',
+          statusCode: StatusCodes.NO_CONTENT,
+        )
+        .then((response) => response.statusCode == StatusCodes.NO_CONTENT);
```

Comment:
Most of the changes will look like this. Update the return type to `Future<bool>`, change the await to a return, and chain a `.then` that will return true on success.